pkgs/development/compilers/terra/default.nix

```diff
@@ -1,61 +1,65 @@
 { lib, stdenv, fetchurl, fetchFromGitHub
-, llvmPackages, ncurses, lua
+, llvmPackages, ncurses, cmake, libxml2
 }:
 
 let
-  luajitArchive = "LuaJIT-2.0.5.tar.gz";
+  luajitRev = "9143e86498436892cb4316550be4d45b68a61224";
+  luajitArchive = "LuaJIT-${luajitRev}.tar.gz";
   luajitSrc = fetchurl {
```

Comment:
Please use fetchFromGitHub.

Reply:
The tarball is part of luring cmake out of download in configure-time.

https://github.com/terralang/terra/blob/release-1.0.0-beta3/cmake/Modules/GetLuaJIT.cmake#L66

Reply:
I don't understand your sentence.

Reply:
The CMake build scripts for Terra are extremely cursed, and have a built-in assumption that they manually download, unpack, and build LuaJIT. My teammates and I are working on correcting that upstream in Terra, but until that is done you're not going to get a deterministic build out of Terra without making Nix do the LuaJIT download on CMake's behalf. 

In short, what this does is deterministically download the exact tarball that Terra's `GetLuaJIT.cmake` expects to acquire, and allow things to proceed normally from there.

There is no way to use `fetchFromGitHub` here without more significant rewriting to the CMake scripts (and that is a long and bloody work in progress).

Reply:
AFAIK the `/archive/` links from GitHub can't be well used with plain `fetchurl` because of not having a longterm stable hash.